File: venda.c
#include <stdio.h>
#include "venda.h"

Venda initVenda() {
    Venda v = malloc(sizeof(struct venda));
    v->produto = malloc(sizeof(char *));
    v->cliente = malloc(sizeof(char *));
    v->preco = 0.0f;
    v->quantidade = 0;
    v->epoca = 'N';
    v->mes = 0;
    v->filial = 0;
    return v;
}

void setVenda(Venda v, char **partes) {
    v->produto = strdup(partes[0]);
    v->preco = atof(partes[1]);
    v->quantidade = atoi(partes[2]);
    v->epoca = partes[3][0];
    v->cliente = strdup(partes[4]);
    v->mes = atoi(partes[5]);
    v->filial = atoi(partes[6]);
}

void showVenda(Venda v) {
    printf("%s %f %i %c %s %i %i\n",v->produto,v->preco,v->quantidade,v->epoca,v->cliente,v->mes,v->filial);
}

File: ProjJava/src/Venda.java
public class Venda {
    private String produto;
    private float preco;
    private int quantidade;
    private char epoca;
    private String cliente;
    private int mes;
    private int filial;

    public Venda(String produto, float preco, int quantidade, char epoca, String cliente, int mes, int filial) {
        this.produto = produto;
        this.preco = preco;
        this.quantidade = quantidade;
        this.epoca = epoca;
        this.cliente = cliente;
        this.mes = mes;
        this.filial = filial;
    }

    public String getProduto() {
        return produto;
    }

    public void setProduto(String produto) {
        this.produto = produto;
    }

    public float getPreco() {
        return preco;
    }

    public void setPreco(float preco) {
        this.preco = preco;
    }

    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }

    public char getEpoca() {
        return epoca;
    }

    public void setEpoca(char epoca) {
        this.epoca = epoca;
    }

    public String getCliente() {
        return cliente;
    }

    public void setCliente(String cliente) {
        this.cliente = cliente;
    }

    public int getMes() {
        return mes;
    }

    public void setMes(int mes) {
        this.mes = mes;
    }

    public int getFilial() {
        return filial;
    }

    public void setFilial(int filial) {
        this.filial = filial;
    }

}